store/types/pruning.go

```diff
@@ -2,18 +2,45 @@ package types
 
 // PruningStrategy specifies how old states will be deleted over time where
 // keepRecent can be used with keepEvery to create a pruning "strategy".
+// CONTRACT: snapshotEvery is a multiple of keepEvery
 type PruningOptions struct {
-	keepRecent int64
-	keepEvery  int64
+	keepRecent    int64
+	keepEvery     int64
+	snapshotEvery int64
 }
 
-func NewPruningOptions(keepRecent, keepEvery int64) PruningOptions {
+func NewPruningOptions(keepRecent, keepEvery, snapshotEvery int64) PruningOptions {
```

Comment:
I would remove this helper. Having 3 arguments of the same type, makes it hard to follow the meaning and easy to mix them up. 2 is bad enough, for 3, just make the struct members public (I see no reason not to)

Reply:
ditto -- remove in favor of having the end-user create a struct literal.